main.bal

```diff
@@ -875,6 +918,19 @@ service / on new http:Listener(serverPort) {
 
 # Start all ResultWatchers from their DB entries.
 public function main() {
+    // insert preset plugin runners and plugins into database
+    transaction {
+        foreach string pRunner in preconfiguredPluginRunners {
+            _ = check database:addPluginEndpoint({url: pRunner, 'type: "PluginRunner"});
```

Comment:
how does this work on the second start? Since the database should ensure uniqueness of the URLs this will fail with an error at the first endpoint that is already in the database which will cancel the whole transaction. Is this the desired behaviour or should newly added URLs always be added to the DB?